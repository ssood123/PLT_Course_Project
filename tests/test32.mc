function void main()
{
  float f;
  bool b;
  f = -72.8;
  b = (1 < 7);
  printf(f);
  printb(b);

}
