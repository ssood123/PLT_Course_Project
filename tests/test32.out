; ModuleID = 'integraph'
source_filename = "integraph"

@fmt = private unnamed_addr constant [4 x i8] c"%d\0A\00", align 1
@fmt.1 = private unnamed_addr constant [4 x i8] c"%s\0A\00", align 1
@fmt.2 = private unnamed_addr constant [4 x i8] c"%g\0A\00", align 1

declare i32 @printf(i8*, ...)

declare i32 @printbig(i32)

define void @main() {
entry:
  %f = alloca double, align 8
  %b = alloca i1, align 1
  store double -7.280000e+01, double* %f, align 8
  store i1 true, i1* %b, align 1
  %f1 = load double, double* %f, align 8
  %printf = call i32 (i8*, ...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @fmt.2, i32 0, i32 0), double %f1)
  %b2 = load i1, i1* %b, align 1
  %printf3 = call i32 (i8*, ...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @fmt, i32 0, i32 0), i1 %b2)
  ret void
}
